gufe/components/explicitmoleculecomponent.py

```diff
@@ -68,16 +68,7 @@ def __init__(self, rdkit: RDKitMol, name: str = ""):
             )
 
         self._rdkit = rdkit
-        self._hash = self._hashmol(name=name)
-
-    def __hash__(self):
-        return hash(self._hash)
-
-    def __eq__(self, other):
-        return hash(self) == hash(other)
-
-    def _hashmol(self, name):
-        return hashmol(self._rdkit, name=name)
+        self._hash = hashmol(self._rdkit, name=name)
```

Comment:
For this to be used by the gufe machinery, shouldn't it be called _key? Either way, we can probably remove the hashmol concept completely and replace with the hash of the to_dict string.

Reply:
We're not using this for hashing. At all. I switched it to gufe `to_dict`-based things in [ce79601](https://github.com/OpenFreeEnergy/gufe/pull/69/commits/ce79601041cf85af5d14465b1de2bd0840f8dea3).

I didn't remove/rename `hashmol` just because I was trying to keep the PR small. I'll do that now.